MemoryManagement/src/memorymanagementAlgorithm/Void.java

```diff
@@ -21,55 +28,83 @@ we have intialy void (free locations)= base = 0 limit 1000
 
 but if it's from 500 to 600 and from 900 to 1000
 we will add the segments easy as 2 free segments
+
+
+ */
+/**
+ *
+ * @author ahmed
  */
 public class Void {
+//   Collections.sort(xx,(a, b) -> a.getBase() < b.getBase() ? -1 : a.getBase() == b.getBase() ? 0 : 1);
 
     private Color color = Color.BLACK;
-    private Vector<Segment> void_vector;
 
+    private Vector<Segment> void_vector;
+    private char condtion = 'b';// b = sort on base,  h=sort on larger at top , 

```

Comment:
No initialize for variable here. Initialization in constructors unless static variables.

Reply:
fixed